### PR TITLE
Add socks proxy options

### PR DESCRIPTION
**Pull request by fabre-thibaud**
Adds options to route traffic through a SOCKS proxy:

- `inputSocksProxy`: defines the proxy host for input transport
- `inputSocksPort`: defines the proxy port for input transport
- `outputSocksProxy`: defines the proxy host for input transport
- `outputSocksPort`: defines the proxy port for input transport

### CODE REVIEW

File: bin/elasticdump [hunk 1]
```diff
@@ -66,7 +66,11 @@ const defaults = {
   'output-cert': null,
   'output-key': null,
   'output-pass': null,
-  'output-ca': null
+  'output-ca': null,
+  inputSocksProxy: null,
```
Comment (ferronrsmith) [inline bin/elasticdump]: change to `input-socks-proxy`, `input-socks-port`, etc

File: lib/transports/elasticsearch.js [hunk 2]
```diff
@@ -51,6 +63,22 @@ class elasticsearch {
     this.baseRequest = request.defaults(defaultOptions)
   }
 
+  applyProxy (url, proxyHost, proxyPort) {
+    let reqAgent = agent
+
+    if (url.substring(0, 5).toLowerCase() === 'https') {
```
Comment (ferronrsmith) [inline lib/transports/elasticsearch.js]: I'd prefer if u use the url object to do the validation here.

```javascript
const myURL = new URL(_url);
if(_url.protocol === 'https:') {}
```

**Comment by ferronrsmith** [comment]
just those 2 changes and I'll merge. Thanks for your contribution !

**Comment by fabre-thibaud** [comment]
Thanks for the merge :+1: !

**Comment by ferronrsmith** [comment]
Thanks for contributing 🥇

**Pull request merged**